src/chatTab.ts

```diff
@@ -234,8 +234,8 @@ export class ChatTab {
         return new Promise<void>((resolve) => {
             const disposables: vscode.Disposable[] = [];
             const restore = (event: { type: string }) => {
+                console.log('----- RESTORE CHAT EVENT',event);
```

Comment:
Okay, so the "READY" message has a `payload` property and that payload has an `id` property.

The id is for the chat that sent the ready message (it's for handling multiple chats).

replace `id: this.chat_id,` on 242 with `event.payload.id ?? this.chat_id` and you should be able to remove the `await new Promise((resolve) => setTimeout(resolve, 250)` in `src/quickProvider.ts`